src/UnitTest/Python/test_core.py

```diff
@@ -0,0 +1,254 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+# ----------------------------------------------------------------------------
+
+import open3d as o3d
+import numpy as np
+import torch
+import torch.utils.dlpack
+import time
+import pytest
+
+
+def list_devices():
+    devices = [o3d.Device("CPU:" + str(0))]
+    if (o3d.cuda.device_count() != torch.cuda.device_count()):
+        raise RuntimeError(
+            "o3d.cuda.device_count() != torch.cuda.device_count(), "
+            f"{o3d.cuda.device_count()} != {torch.cuda.device_count()}")
+    for i in range(o3d.cuda.device_count()):
+        devices.append(o3d.Device("CUDA:" + str(i)))
+    return devices
+
+
+def test_dtype():
+    dtype = o3d.Dtype.Int32
+    assert o3d.DtypeUtil.byte_size(dtype) == 4
+    assert o3d.DtypeUtil.to_string(dtype) == "Int32"
+
+
+def test_device():
+    device = o3d.Device()
+    assert device.get_type() == o3d.Device.DeviceType.CPU
+    assert device.get_id() == 0
+
+    device = o3d.Device("CUDA", 1)
+    assert device.get_type() == o3d.Device.DeviceType.CUDA
+    assert device.get_id() == 1
+
+    device = o3d.Device("CUDA:2")
+    assert device.get_type() == o3d.Device.DeviceType.CUDA
+    assert device.get_id() == 2
+
+    assert o3d.Device("CUDA", 1) == o3d.Device("CUDA:1")
+    assert o3d.Device("CUDA", 1) != o3d.Device("CUDA:0")
+
+    assert o3d.Device("CUDA", 1).to_string() == "CUDA:1"
+
+
+def test_size_vector():
+    # List
+    sv = o3d.SizeVector([-1, 2, 3])
+    assert f"{sv}" == "{-1, 2, 3}"
+
+    # Tuple
+    sv = o3d.SizeVector((-1, 2, 3))
+    assert f"{sv}" == "{-1, 2, 3}"
+
+    # Numpy 1D array
+    sv = o3d.SizeVector(np.array([-1, 2, 3]))
+    assert f"{sv}" == "{-1, 2, 3}"
+
+    # Empty
+    sv = o3d.SizeVector()
+    assert f"{sv}" == "{}"
+    sv = o3d.SizeVector([])
+    assert f"{sv}" == "{}"
+    sv = o3d.SizeVector(())
+    assert f"{sv}" == "{}"
+    sv = o3d.SizeVector(np.array([]))
+    assert f"{sv}" == "{}"
+
+    # Automatic int casting (not rounding to nearest)
+    sv = o3d.SizeVector((1.9, 2, 3))
+    assert f"{sv}" == "{1, 2, 3}"
+
+    # Automatic casting negative
+    sv = o3d.SizeVector((-1.5, 2, 3))
+    assert f"{sv}" == "{-1, 2, 3}"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=4843493)